common/src/main/java/com/wynntils/mc/mixin/MinecraftMixin.java

```diff
@@ -26,6 +28,13 @@ private void setScreenPostPost(Screen screen, CallbackInfo ci) {
     @Inject(method = "tick", at = @At("HEAD"))
     private void tickPre(CallbackInfo ci) {
         EventFactory.onTickStart();
+
+        Queue<Runnable> runnableQueue = MinecraftSchedulerManager.getQueue();
```

Comment:
I'm thinking this could be done by the MinecraftScheduleManager itself, by listening to onTickStart event? 

Otoh... That makes it always call, even if it does not have something to do. Was this the reason?

Reply:
Reason was I needed to run this on  the game thread. Now I realize I could have used an event in the manager as well. Maybe it is better to inject this here

Reply:
If there's anything that worries me about performance, then it's stuff done for each tick, or for each render call. 

So design-wise I think the manager should have had an event listener. But, that will make the event handler called at each tick, regardless of if it has anything to do (as it will be most time) or not. That could be mitigated by registering the manager if a runnable is enqueued, and unregistering it when the queue gets empty. But that is also overhead, and will not really fit our design either. 

So I guess this is for the best (even if untested assumptions about performance is the "premature optimization" that is the root of all evil...). If anything, the code should be broken out into a separate method in MinecraftSchedulerManager like `onTick()`, so we don't have to expose the queue and the internal logic.